### DISCUSSION TITLE
Handling of optional values and exceptions with pattern matching

### DISCUSSION THREAD

**Discussion by ozturkberkay** [ValeLang/Vale#513] [Q&A] [answered]
Hi,

I was reading the documentation and I saw that the `match` statement is a planned feature. I was wondering if there are any plans for introducing something similar to `Option` and `Result` enums in Rust. I'm curious how empty values and exceptions will be handled in Vale.

Thank you

**Comment by Verdagon** [comment, marked as answer]
Hey! Vale currently has Result and Opt, a panic function, and no exceptions. It's most like Rust in this particular aspect.
